ugvc/mrd/balanced_strand_utils.py

```diff
@@ -0,0 +1,992 @@
+                    )
+                else:
+                    record.info[HistogramColumnNames.STRAND_RATIO_START.value] = np.nan
+                record.info[HistogramColumnNames.STRAND_RATIO_CATEGORY_START.value] = get_annotation(
+                    record.info[HistogramColumnNames.STRAND_RATIO_START.value],
+                    self.sr_lower,
+                    self.sr_upper,
+                )
+            if self.adapter_version in (
+                BalancedStrandAdapterVersions.LA_v5and6.value,
+                BalancedStrandAdapterVersions.LA_v6.value,
+            ):  # LA_v6 has end tags
+                # assign to simple variables for readability
+                T_hmer_end = balanced_tags[TrimmerSegmentTags.T_HMER_END.value]
+                A_hmer_end = balanced_tags[TrimmerSegmentTags.A_HMER_END.value]
+                # determine ratio and category
+                tags_sum_end = T_hmer_end + A_hmer_end
+                # determine if read end was reached
+                is_end_reached = (
+                    balanced_tags[TrimmerSegmentTags.NATIVE_ADAPTER.value] >= 1
+                    or balanced_tags[TrimmerSegmentTags.STEM_END.value] >= self.min_stem_end_matched_length
+                )
+                record.info[HistogramColumnNames.STRAND_RATIO_END.value] = np.nan
+                if not is_end_reached:
+                    record.info[
+                        HistogramColumnNames.STRAND_RATIO_CATEGORY_END.value
+                    ] = BalancedCategories.END_UNREACHED.value
+                else:
+                    if self.min_total_hmer_lengths_in_tags <= tags_sum_end <= self.max_total_hmer_lengths_in_tags:
+                        record.info[HistogramColumnNames.STRAND_RATIO_END.value] = T_hmer_end / (
+                            T_hmer_end + A_hmer_end
+                        )
+                    record.info[HistogramColumnNames.STRAND_RATIO_CATEGORY_END.value] = get_annotation(
+                        record.info[HistogramColumnNames.STRAND_RATIO_END.value],
+                        self.sr_lower,
+                        self.sr_upper,
+                    )  # this works for nan values as well - returns UNDETERMINED
+                print(
+                    is_end_reached,
+                    balanced_tags[TrimmerSegmentTags.NATIVE_ADAPTER.value],
+                    balanced_tags[TrimmerSegmentTags.STEM_END.value],
+                    tags_sum_end,
+                    (self.min_total_hmer_lengths_in_tags <= tags_sum_end <= self.max_total_hmer_lengths_in_tags),
+                    balanced_tags,
+                )
+                print(record)
+            records_out[j] = record
+
+        return records_out
+
+
+# Display defaults
+STRAND_RATIO_AXIS_LABEL = "LIG/HYB strands ratio"
+
+# Misc
+balanced_category_list = [v.value for v in BalancedCategories.__members__.values()]
+supported_adapter_versions = BalancedStrandAdapterVersions.__members__.values()
```

Comment:
ditto

Reply:
and capitalize for constant?

Reply:
They are defined by the classes above so they can't be in the top of the file. They're not exactly contants either, not sure what the right convention is...